### PR TITLE
repos: Clarify usage and which RHEL versions are available in repos

### PR DESCRIPTION
**Pull request by travier**
For RHCOS, we want to be able to have a stable name for each RHEL versions available from a repo to help us during migrations from one RHEL minor to the next. Thus we will no longer used the un-versioned `rhel8` repos and use minor versioned ones: `rhel86`, `rhel90`.

Whether or not the URLs used for the repos are minor versioned is a different issue that we are waiting for confirmation. See https://github.com/openshift/ocp-build-data/pull/1576#issuecomment-1123901440 & https://github.com/openshift/release/pull/28503#issuecomment-1124188493

Keep the un-versioned one `rhel8` for compatibility for other potential users and make it use the latest released RHEL.

### CODE REVIEW

**Comment by travier** [comment]
CC @miabbott @cgwalters @yselkowitz

**Comment by miabbott** [comment]
Looks great!  Thanks for doing this.

/approve

File: core-services/release-controller/_repos/ocp-4.11-rhel85.repo [hunk 1]
```diff
@@ -55,46 +55,10 @@ gpgkey = https://mirror.ops.rhcloud.com/libra/keys/RPM-GPG-KEY-redhat-release ht
 
 [rhel-8-advanced-virt]
 name = rhel-8-advanced-virt
-baseurl = https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/advanced-virt/os/
+baseurl = https://cdn.redhat.com/content/eus/rhel8/x86_64/advanced-virt/os/
```
Comment (travier) [inline core-services/release-controller/_repos/ocp-4.11-rhel85.repo:58]: I've realized that this is maybe not what we want right now as we have: https://github.com/openshift/os/pull/796.
But I'm not sure. Should we also revert that until we go back to 8.6?

Comment (miabbott) [reply core-services/release-controller/_repos/ocp-4.11-rhel85.repo:58]: Eduardo has a PR to remove RHEL-AV as part of 8.6 - https://github.com/openshift/release/pull/26811

**Comment by travier** [comment]
Updated to use RHEL 8.6 EUS everywhere.

**Comment by miabbott** [comment]
Since we are doing clean up in this PR, couple of questions:

- do we still need an `aarch64` version of the 4.11 repo: `ocp-4.11-rhel8-aarch64.repo`
- should the unversioned rhel8 repo (`ocp-4.11-rhel8.repo`) change the path in the `baseurl` to be `content/dist/rhel8/8/x86_64/appstream/os/`?  Users that need specific versions of RHEL should be able to reference one of the explicitly versioned repos.

**Comment by yselkowitz** [comment]
Yes we still need aarch64 repo files because the "main" ones hard-code `x86_64`.

**Comment by travier** [comment]
> * should the unversioned rhel8 repo (`ocp-4.11-rhel8.repo`) change the path in the `baseurl` to be `content/dist/rhel8/8/x86_64/appstream/os/`?  Users that need specific versions of RHEL should be able to reference one of the explicitly versioned repos.

I think we still want that repo to be on the "correct" RHEL version for a given OCP release. I kept it for consumers that want the latest valid RHEL for a given OCP release without caring about the version itself.

**Comment by travier** [comment]
We might however want to remove some RHCOS specific repos from those files. I'll try that.

**Comment by travier** [comment]
Updated this PR to focus on cleanups and the switch to RHEL 8.6 in 4.11 and later

**Pull request merged**